hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/FileRangeImpl.java

```diff
@@ -15,11 +15,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.hadoop.fs;
+package org.apache.hadoop.fs.impl;
 
 import java.nio.ByteBuffer;
 import java.util.concurrent.CompletableFuture;
 
+import org.apache.hadoop.fs.FileRange;
+
```

Comment:
1. declare as @Private just to make clear it should not be directly created...and say this in the javadoc
2. should offset and length be final? or are the setters needed?

Reply:
1. done.
2. Yeah setter for length is required during merging.